lib/wickedbase.pm

```diff
@@ -233,9 +233,18 @@ sub ping_with_timeout {
     while ($timeout > 0) {
         if (script_run($ping_command) == 0) {
             if ($args{count_success} > 1) {
-                my $cnt = $args{count_success} - 1;
+                my $cnt       = $args{count_success} - 1;
+                my $threshold = 50;
                 $ping_command =~ s/\s-c\s1\s+/ -c $cnt /;
-                validate_script_output($ping_command, sub { /\s+0% packet loss/ });
+                my $ping_out = script_output($ping_command, proceed_on_failure => 1);
+                $ping_out =~ /, (\d{1,3})% packet loss/;
+                #we treat interface in workable state if it manage to echo more than half packets
```

Comment:
I wouldn't state this without a good reason. Why 50%?  I would vote for having threshold as input parameter and let the user of this function decide

Reply:
fixed